lib/src/entry/ongoing_query.dart

```diff
@@ -15,87 +15,89 @@ class OngoingQuery {
-        }
+      if (added.isNotEmpty) {
+        forEachResponse((response) {
+          response.updateStream(
+            [],
+            columns: knownColumns
+              .map((name) => new TableColumn(name, "dynamic"))
+              .toList()
+          );
+        });
       }
+    }
 
-      String path = update.id;
-
-      if (!rows.containsKey(path)) {
-        if (!update.remove) {
-          var row = rows[path] = new QueryTableRow(update.values);
-          row.id = rows.length - 1;
-
-          forEachResponse((response) {
-            response.updateStream(
-              [row.format(knownColumns)],
-              meta: {
-                "mode": "append"
-              }
-            );
-          });
-        }
-      } else {
-        if (update.remove) {
-          QueryTableRow row = rows.remove(path);
-          for (QueryTableRow m in rows.values) {
-            if (m.id > row.id) {
-              m.id--;
+    String path = update.id;
+
+    if (!rows.containsKey(path)) {
+      if (!update.remove) {
+        var row = rows[path] = new QueryTableRow(update.values);
+        row.id = rows.length - 1;
+
+        forEachResponse((response) {
+          response.updateStream(
+            [row.format(knownColumns)],
+            meta: {
+              "mode": "append"
             }
+          );
+        });
+      }
+    } else {
+      if (update.remove) {
+        QueryTableRow row = rows.remove(path);
+        for (QueryTableRow m in rows.values) {
+          if (m.id > row.id) {
+            m.id--;
           }
-
-          forEachResponse((response) {
-            response.updateStream(
-              rows.values.where((r) {
-                return r.id >= row.id;
-              }).map((row) => row.format(knownColumns)).toList(),
-              meta: {
-                "modify": "replace ${row.id}-${rows.length}"
-              }
-            );
-          });
-        } else {
-          QueryTableRow row = rows[path];
-          row.values.addAll(update.values);
-          forEachResponse((response) {
-            response.updateStream(
-              [row.format(knownColumns)],
-              meta: {
-                "modify": "replace ${row.id}-${row.id}"
-              }
-            );
-          });
         }
-      }
 
-      if (forceRefresh) {
         forEachResponse((response) {
-          sendToResponse(response);
+          response.updateStream(
+            rows.values.where((r) {
+              return r.id >= row.id;
+            }).map((row) => row.format(knownColumns)).toList(),
+            meta: {
+              "modify": "replace ${row.id}-${rows.length}"
+            }
+          );
+        });
+      } else {
+        QueryTableRow row = rows[path];
+        row.values.addAll(update.values);
+        forEachResponse((response) {
+          response.updateStream(
+            [row.format(knownColumns)],
+            meta: {
+              "modify": "replace ${row.id}-${row.id}"
```

Comment:
is that `"replace ${row.id}-${row.id}"` correct? Kinda looks like a copy-paste mistake, I'm not sure what this means anyway.

Reply:
Replace supports ranges, so this is correct. It tells it to replace only one row.